### PR TITLE
Add Test Proxy in Mixed Reality Authentication SDK

### PR DESCRIPTION
**Pull request by g2vinay**
Integrates Test Proxy in Mixed Reality Authentication SDK

### CODE REVIEW

File: sdk/mixedreality/azure-mixedreality-authentication/src/test/java/com/azure/mixedreality/authentication/MixedRealityStsClientTestBase.java [hunk 1]
```diff
@@ -40,13 +43,21 @@ HttpPipeline getHttpPipeline(HttpClient httpClient) {
         final List<HttpPipelinePolicy> policies = new ArrayList<>();
         policies.add(new BearerTokenAuthenticationPolicy(credential, authenticationScope));
 
-        if (getTestMode() == TestMode.RECORD) {
+        if (interceptorManager.isRecordMode()) {
```
Comment (samvaity) [inline sdk/mixedreality/azure-mixedreality-authentication/src/test/java/com/azure/mixedreality/authentication/MixedRealityStsClientTestBase.java:46]: sanitizers are to be added in both playback and record mode to match correctly with redacted content.

Comment (g2vinay) [reply sdk/mixedreality/azure-mixedreality-authentication/src/test/java/com/azure/mixedreality/authentication/MixedRealityStsClientTestBase.java:48]: I assume that's applicable only if we're redacting RequestUri and Headers ? as those are what's used to match the request ?
Else, I should see failures in CI.

File: sdk/mixedreality/azure-mixedreality-authentication/src/test/java/com/azure/mixedreality/authentication/MixedRealityStsClientTestBase.java [hunk 2]
```diff
@@ -40,13 +45,21 @@ HttpPipeline getHttpPipeline(HttpClient httpClient) {
...
+            List<TestProxySanitizer> customSanitizers = new ArrayList<>();
+            customSanitizers.add(new TestProxySanitizer("$..AccessToken", null, "eyJhbGciOiJub25lIiwidHlwIjoiSldUIn0.eyJlbWFpbCI6IkJvYkBjb250b3NvLmNvbSIsImdpdmVuX25hbWUiOiJCb2IiLCJpc3MiOiJodHRwOi8vRGVmYXVsdC5Jc3N1ZXIuY29tIiwiYXVkIjoiaHR0cDovL0RlZmF1bHQuQXVkaWVuY2UuY29tIiwiaWF0IjoiMTYwNzk3ODY4MyIsIm5iZiI6IjE2MDc5Nzg2ODMiLCJleHAiOiIxNjA3OTc4OTgzIn0.", TestProxySanitizerType.BODY_KEY));
```
Comment (samvaity) [inline sdk/mixedreality/azure-mixedreality-authentication/src/test/java/com/azure/mixedreality/authentication/MixedRealityStsClientTestBase.java:50]:
```suggestion
            customSanitizers.add(new TestProxySanitizer("$..AccessToken", null, "REDACTED", TestProxySanitizerType.BODY_KEY));
```

File: sdk/mixedreality/azure-mixedreality-authentication/src/test/java/com/azure/mixedreality/authentication/MixedRealityStsClientTestBase.java (hunk 1 above)
Comment (samvaity) [reply sdk/mixedreality/azure-mixedreality-authentication/src/test/java/com/azure/mixedreality/authentication/MixedRealityStsClientTestBase.java:48]: We pull up the match using the request URI but after that we just match everything, I see you have added BodilessMatcher it coul be why we don't see the match errors.

File: sdk/mixedreality/azure-mixedreality-authentication/src/test/java/com/azure/mixedreality/authentication/MixedRealityStsClientTestBase.java (hunk 2 above)
Comment (joshfree) [inline sdk/mixedreality/azure-mixedreality-authentication/src/test/java/com/azure/mixedreality/authentication/MixedRealityStsClientTestBase.java:50]: gave feedback offline; add comment make this more obvious that it's an invalid dummy token

**Review by joshfree — APPROVED** [review]

File: sdk/mixedreality/azure-mixedreality-authentication/src/test/java/com/azure/mixedreality/authentication/MixedRealityStsClientTestBase.java (hunk 2 above)
Comment (g2vinay) [reply sdk/mixedreality/azure-mixedreality-authentication/src/test/java/com/azure/mixedreality/authentication/MixedRealityStsClientTestBase.java]: This is an invalid token needed for tests validation.
It replaces the real token in response with the dummy token.

**Pull request merged**